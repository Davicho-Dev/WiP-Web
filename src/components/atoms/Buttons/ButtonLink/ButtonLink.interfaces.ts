export interface IButtonLinkProps {
	label: string
	className?: string
	type?: 'button' | 'submit' | 'reset'
	onClick?: () => void
	disabled?: boolean
}
