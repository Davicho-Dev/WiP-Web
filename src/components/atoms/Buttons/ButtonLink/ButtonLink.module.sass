.button
  @apply w-fit h-fit bg-transparent text-primary underline
