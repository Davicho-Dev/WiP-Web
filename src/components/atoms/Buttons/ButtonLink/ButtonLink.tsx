import { IButtonLinkProps } from './ButtonLink.interfaces'

import styles from './ButtonLink.module.sass'

export const ButtonLink = ({
	className,
	label,
	type,
	onClick,
}: IButtonLinkProps): JSX.Element => {
	return (
		<button
			className={`${styles.button} ${className ?? ''}`}
			onClick={onClick}
			type={type}
		>
			{label}
		</button>
	)
}
