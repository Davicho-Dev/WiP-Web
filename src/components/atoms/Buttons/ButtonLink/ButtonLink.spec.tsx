import { render } from '@testing-library/react'

import { ButtonLink } from '../ButtonLink'

describe('ButtonLink test suite', () => {
	it('Should render correctly', () => {
		const label = 'Button Link'

		const { getByText } = render(<ButtonLink {...{ label }} />)

		expect(getByText(label)).toMatchSnapshot()
	})
})
