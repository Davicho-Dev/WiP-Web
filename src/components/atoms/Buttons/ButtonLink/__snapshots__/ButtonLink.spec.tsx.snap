// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink test suite Should render correctly 1`] = `
<button
  class="button "
>
  Button Link
</button>
`;
